core/templates/dev/head/pages/library/Library.js

```diff
@@ -113,6 +118,131 @@ oppia.controller('Library', [
+            activityType: activityType,
+            activityId: activityId
+          }));
+      $http.post(addActivityToLearnerPlaylistUrl, {})
+        .then(function(response) {
+          if (response.data.belongs_to_completed_or_incomplete_list) {
+            alertsService.addInfoMessage(
+              'You have already completed or are completing this activity.');
+          } else if (response.data.belongs_to_subscribed_activities) {
+            alertsService.addInfoMessage(
+              'This is present in your creator dashboard');
+          } else if (response.data.playlist_limit_exceeded) {
+            alertsService.addInfoMessage(
+              'Your \'Play Later\' list is full!  Either you can complete ' +
+              'some or you can head to the learner dashboard and remove some.');
+          } else {
+            alertsService.addSuccessMessage(
+              'Succesfully added to your \'Play Later\' list.');
+          }
+        });
+
+      if (activityType == constants.ACTIVITY_TYPE_EXPLORATION) {
+        $scope.learnerDashboardActivityIds.exploration_playlist_ids.push(
+          activityId);
+      } else {
+        $scope.learnerDashboardActivityIds.collection_playlist_ids.push(
+          activityId);
+      }
+    };
+
+    $scope.removeFromLearnerPlaylist = function(
+      activityId, activityType, activityTitle) {
+      $modal.open({
+        templateUrl: 'modals/removeActivity',
+        backdrop: true,
+        resolve: {
+          activityId: function() {
+            return activityId;
+          },
+          activityType: function() {
+            return activityType;
+          },
+          activityTitle: function() {
+            return activityTitle;
+          }
+        },
+        controller: [
+          '$scope', '$modalInstance', '$http', 'UrlInterpolationService',
+          function($scope, $modalInstance, $http, UrlInterpolationService) {
+            $scope.sectionNameI18nId = (
+              'I18N_LEARNER_DASHBOARD_PLAYLIST_SECTION');
+            $scope.activityTitle = activityTitle;
+            var removeFromLearnerPlaylistUrl = (
+              UrlInterpolationService.interpolateUrl(
+                '/learnerplaylistactivityhandler/<activityType>/<activityId>', {
+                  activityType: activityType,
+                  activityId: activityId
+                }));
+
+            $scope.remove = function() {
+              $http['delete'](removeFromLearnerPlaylistUrl);
+              $modalInstance.close();
+            };
+
+            $scope.cancel = function() {
+              $modalInstance.dismiss('cancel');
+            };
+          }
+        ]
+      }).result.then(function() {
+        if (activityType == constants.ACTIVITY_TYPE_EXPLORATION) {
+          var index = (
+            $scope.learnerDashboardActivityIds.exploration_playlist_ids.indexOf(
+              activityId));
+          if (index !== -1) {
```

Comment:
This cannot happen in practice. Just introduced this as an extra check. If this happens it means that the user has clicked on the 'x' button on an activity that does not exist.